Add docs to chat conversation

charlesLoder opened the pull request:
## Description

Adds the ability to add search results back into the conversation history as implemented in the [backend PR](https://github.com/nulib/dc-api-v2/pull/293)

## Summary of changes

- Adds `userDocs` to each `Turn`
- Adds `latestDocs` to the conversation history
- Users can optionally add filtered results to the chat history
- Users can remove results from history
- Users can ask about those docs specifically

## Example

Live: https://preview-5394-docs-to-convo.dc.rdc-staging.library.northwestern.edu/

Ask about a subject and go to "View results"

<img width="1319" alt="Screenshot 2025-03-24 at 4 15 48 PM" src="https://github.com/user-attachments/assets/abd8d858-9a52-4e89-a4f0-8724d59f96af" />

Filter results — will add a checkbox for "Add results to conversation". If checkbox is deselected, the user just returns to the conversation as normal

<img width="1638" alt="Screenshot 2025-03-24 at 4 16 18 PM" src="https://github.com/user-attachments/assets/367f414b-7a7a-4c66-bd22-74633fc9abda" />

Click "Back to the conversation" and return to the chat with a new interstitial. At this point nothing has been submitted to the API. The user can remove the docs if they desire. Note that the new chat prompt is "Ask about these documents".

<img width="1468" alt="Screenshot 2025-03-24 at 4 16 39 PM" src="https://github.com/user-attachments/assets/be9a72ac-f725-4dfd-bf63-bc549628ea1b" />

Ask a question about the selected documents:

<img width="1297" alt="Screenshot 2025-03-24 at 4 17 34 PM" src="https://github.com/user-attachments/assets/0855efa4-4b70-47b1-9af8-488597546b1b" />


## Notes

- There is a discrepancy where the `retrieve_documents` tool message doesn't seem to search for all the documents (or it isn't indicating that it is)

## Loose ends

Just jotting down things to come back to, no action needed now:
- docs in Feedback?
- naming conventions — docs vs results vs ???
- filter styling is still cut off

mathewjordan commented on components/Chat/Response/InterstitialDocuments.tsx:
```diff
@@ -0,0 +1,85 @@
+          <StyledInterstitialIcon>
+            <IconSparkles />
+          </StyledInterstitialIcon>
+          Selected documents ({documents.length}):
```
I wonder if we can just leave off documents.length here? keep it vague?

mathewjordan commented on components/Chat/Response/InterstitialDocuments.tsx:
```diff
@@ -0,0 +1,85 @@
+  };
+  return (
+    <>
+      <StyledInterstitialWrapper
```
Thoughts, perhaps this can be part of a new issue/iteration.

My perspective is that the wrapper isn't obviously part of the question because it resembles the AI Search Results interstitial.

- Can we wrap this in a color matching our Question bubble to make it more obvious that it is user created? 
- Maybe even NOT making it full width, 2/3 or 61.8%? Aligning right? And/or bringing it closer to the textarea, or somehow wrapper by the textarea.
- Smaller thumbnails? Not showing all, but a stack of them. Obviously this means we need a new component.

mathewjordan reviewed (approved):
I have a lot of thoughts but I don't think it fundametally gets in the way of the purpose of the issue. We can ask questions! We have context! Please take my ideas with a grain of salt, and maybe we can pair to experiment and refine early in the next cycle.

charlesLoder commented:
@mathewjordan see latest changes:

- removed the length in the `InterstitialDocuments` message
- removed the `Retrieving documents` interstitial
- add small thumbnails to the question when a user submit a question with docs

I didn't yet make the thumbnails in the `InterstitialDocuments` smaller.

I'm trying to keep the distinction between when a user navigates back to the chat and sees thumbnails (meaning they haven't actually asked about those docs yet) and between when they actually ask a question with docs attached.

It's still not there, but hopefully on the right track
